### PR TITLE
Aria expanded collapsible content

### PR DESCRIPTION
**Pull request by discodavey**

### CODE REVIEW

File: test/articledownloadlinkslist.spec.js [hunk 1]
```diff
@@ -11,10 +17,39 @@ describe('An ArticleDownloadLinksList Component', function () {
...
+  });
+
+  describe('The open method', function () {
+
+    it('Sets aria-expanded attribute to "true"', function () {
+      articleDownloadLinksList.open();
+      expect(articleDownloadLinksList.$elm.setAttribute.calledWithExactly('aria-expanded', true)).to.be.true;
+    })
+
+    it('Sets class "visuallyhidden" to be removed', function () {
```
Comment (davidcmoulton) [inline test/articledownloadlinkslist.spec.js]: `'removes the class "visuallyhidden"'` reads better.

File: test/articledownloadlinkslist.spec.js [hunk 2]
```diff
@@ -11,10 +17,39 @@ describe('An ArticleDownloadLinksList Component', function () {
...
-  it('exists', function () {
-    expect(articleDownloadLinksList).to.exist;
+  describe('The close method', function () {
+
+    it('Sets aria-expanded attribute to "false"', function () {
+      articleDownloadLinksList.close();
+      expect(articleDownloadLinksList.$elm.setAttribute.calledWithExactly('aria-expanded', false)).to.be.true;
+    });
+
+    it('Sets class "visuallyhidden" to be added', function () {
```
Comment (davidcmoulton) [inline test/articledownloadlinkslist.spec.js]: `'adds the class "visuallyhidden"'` reads better.

File: test/articledownloadlinkslist.html [hunk 3]
```diff
@@ -27,7 +27,7 @@
 
 
 
-    <div data-behaviour="ArticleDownloadLinksList" id="articleDownloadLinksList">
+    <div data-behaviour="ArticleDownloadLinksList" id="articleDownloadLinksList" aria-expanded="false">
```
[the remark this answers is not in the document]
Comment (discodavey) [reply test/articledownloadlinkslist.html]: Completed

File: test/articledownloadlinkslist.spec.js [hunk 4]
```diff
@@ -11,10 +17,39 @@ describe('An ArticleDownloadLinksList Component', function () {
...
+    spy(articleDownloadLinksList.$elm, 'setAttribute');
+  });
+
+  afterEach(function() {
+    articleDownloadLinksList.$elm.setAttribute.restore();
+  });
+
+  describe('The open method', function () {
+
+    it('Sets aria-expanded attribute to "true"', function () {
```
[the remark this answers is not in the document]
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

File: assets/js/components/ArticleDownloadLinksList.js [hunk 5]
```diff
@@ -71,6 +71,7 @@ module.exports = class ArticleDownloadLinksList {
    */
   open() {
     this.$elm.classList.remove('visuallyhidden');
+    this.$elm.setAttribute('aria-expanded', true);
```
[the remark this answers is not in the document]
Comment (discodavey) [reply assets/js/components/ArticleDownloadLinksList.js]: Completed

File: assets/js/components/ArticleDownloadLinksList.js [hunk 6]
```diff
@@ -90,6 +91,7 @@ module.exports = class ArticleDownloadLinksList {
    */
   close() {
     this.$elm.classList.add('visuallyhidden');
+    this.$elm.setAttribute('aria-expanded', false);
```
[the remark this answers is not in the document]
Comment (discodavey) [reply assets/js/components/ArticleDownloadLinksList.js]: Completed

File: test/articledownloadlinkslist.spec.js [hunk 7]
```diff
@@ -11,10 +17,39 @@ describe('An ArticleDownloadLinksList Component', function () {
...
+      articleDownloadLinksList.open();
+      expect(articleDownloadLinksList.$elm.setAttribute.calledWithExactly('aria-expanded', true)).to.be.true;
```
[the remark this answers is not in the document]
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

File: test/articledownloadlinkslist.spec.js [hunk 8]
```diff
@@ -11,10 +17,39 @@ describe('An ArticleDownloadLinksList Component', function () {
...
+
   });
 
-  it('exists', function () {
-    expect(articleDownloadLinksList).to.exist;
+  describe('The close method', function () {
+
+    it('Sets aria-expanded attribute to "false"', function () {
+      articleDownloadLinksList.close();
+      expect(articleDownloadLinksList.$elm.setAttribute.calledWithExactly('aria-expanded', false)).to.be.true;
```
[the remark this answers is not in the document]
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

File: test/articledownloadlinkslist.spec.js (hunk 1 above)
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

File: test/articledownloadlinkslist.spec.js (hunk 2 above)
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

**Comment by discodavey** [comment]
Review changes done

File: test/articledownloadlinkslist.spec.js [hunk 9]
```diff
@@ -11,10 +17,39 @@ describe('An ArticleDownloadLinksList Component', function () {
...
+  afterEach(function() {
+    articleDownloadLinksList.$elm.setAttribute.restore();
+  });
+
+  describe('The open method', function () {
+
+    it('sets aria-expanded attribute to "true"', function () {
+      articleDownloadLinksList.open();
+      expect(articleDownloadLinksList.$elm.setAttribute.calledWithExactly('aria-expanded', "true")).to.be.true;
+    })
```
Comment (davidcmoulton) [inline test/articledownloadlinkslist.spec.js]: Missing semi colon.

File: test/articledownloadlinkslist.spec.js [hunk 10]
```diff
@@ -1,4 +1,10 @@
-let expect = chai.expect;
+const chai = require('chai');
+const sinon = require('sinon');
+
+const expect = chai.expect;
+const spy = sinon.spy;
+
+let utils = require('../assets/js/libs/elife-utils')();
```
Comment (davidcmoulton) [inline test/articledownloadlinkslist.spec.js]: This isn't used in this file, so don't need to import it.

File: test/articledownloadlinkslist.spec.js [hunk 11]
```diff
@@ -11,10 +17,39 @@ describe('An ArticleDownloadLinksList Component', function () {
...
+
+    it('adds the class "visuallyhidden"', function () {
+      articleDownloadLinksList.close();
+      expect(articleDownloadLinksList.$elm.classList.contains('visuallyhidden')).to.be.true;
+    });
+
   });
 
-});
+});
```
Comment (davidcmoulton) [inline test/articledownloadlinkslist.spec.js]: Convention is that all files should finish with an empty line, so just add a return here.

File: test/articledownloadlinkslist.spec.js (hunk 9 above)
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

File: test/articledownloadlinkslist.spec.js (hunk 10 above)
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

File: test/articledownloadlinkslist.spec.js (hunk 11 above)
Comment (discodavey) [reply test/articledownloadlinkslist.spec.js]: Completed

**Comment by discodavey** [comment]
All changes done for above

**Pull request merged**